compiler/src/dotty/tools/dotc/typer/Implicits.scala

```diff
@@ -552,6 +552,7 @@ trait Implicits { self: Typer =>
         || (from.tpe isRef defn.NothingClass)
         || (from.tpe isRef defn.NullClass)
         || !(ctx.mode is Mode.ImplicitsEnabled)
+        || from.isInstanceOf[Super]
```

Comment:
This test is pretty cheap compared to the other ones. I'm wondering if it is better moved earlier

Reply:
Ah, I didn't wonder about that... `Super` is also pretty rare so I'm not sure, but https://github.com/scala/scala/commit/bed3304bf86 still puts it early.